Lib/test/test_when.py

```diff
@@ -0,0 +1,117 @@
+import unittest
+from when import *
+
+# Initial test cases for when and cowns
+# Note: no concurrency test yet
+class WhenTest(unittest.TestCase):
+    obj = None
+
+    def setUp(self):
+        makeimmutable(self.obj)
+
+    def test_cown(self):
+        def invalid_assignment1(c):
+            c.value = 42
+        def invalid_assignment2(c):
+            c.f = 42
+        def invalid_assignment3(c):
+            c["g"] = 42
+
+        c = Cown()
+        self.assertRaises(AttributeError, invalid_assignment1, c)
+        self.assertRaises(AttributeError, invalid_assignment2, c)
+        self.assertRaises(TypeError, invalid_assignment3, c)
+        # Cannot access unacquired cown
+        self.assertRaises(RegionError, lambda _ : c.get(), c)
```

Comment:
Looking at this test, I feel like Cowns should have their own Error type, but that can be done in a followup PR

Reply:
That may be right. I was thinking of region errors as "Pyrona errors" for now.